File: scripts/replace-text-slate.py
from pathlib import Path
import re

root = Path('f:/Project/Minisoccer')
patterns = [
    (re.compile(r'text-slate-400'), 'text-[color:var(--muted)]'),
    (re.compile(r'text-slate-300'), 'text-[color:var(--muted)]'),
    (re.compile(r'text-slate-200'), 'text-[color:var(--foreground)]'),
]

files = list(root.rglob('*.tsx')) + list(root.rglob('*.ts')) + list(root.rglob('*.jsx')) + list(root.rglob('*.js'))
updated = 0
for file in files:
    text = file.read_text(encoding='utf-8')
    new = text
    for pat, repl in patterns:
        new = pat.sub(repl, new)
    if new != text:
        file.write_text(new, encoding='utf-8')
        updated += 1
print(f'updated {updated} files')

File: scripts/replace-text-slate.js
const fs = require('fs');
const path = require('path');

const root = path.resolve('f:/Project/Minisoccer');
const patterns = [
  [/text-[color:var(--muted)]/g, 'text-[color:var(--muted)]'],
  [/text-[color:var(--muted)]/g, 'text-[color:var(--muted)]'],
  [/text-[color:var(--foreground)]/g, 'text-[color:var(--foreground)]'],
];

const exts = ['.tsx', '.ts', '.jsx', '.js'];
let updated = 0;

function walk(dir) {
  for (const name of fs.readdirSync(dir, { withFileTypes: true })) {
    const full = path.join(dir, name.name);
    if (name.isDirectory()) {
      walk(full);
      continue;
    }
    if (!exts.includes(path.extname(name.name))) continue;
    let content = fs.readFileSync(full, 'utf8');
    let next = content;
    for (const [pattern, replacement] of patterns) {
      next = next.replace(pattern, replacement);
    }
    if (next !== content) {
      fs.writeFileSync(full, next, 'utf8');
      updated += 1;
    }
  }
}

walk(root);
console.log(`updated ${updated} files`);
